frappe/core/doctype/version/version.py

```diff
@@ -167,3 +168,10 @@ def get_diff(old, new, for_child=False, compare_cancelled=False):
 
 def on_doctype_update():
 	frappe.db.add_index("Version", ["ref_doctype", "docname"])
+
+
+@frappe.whitelist()
+def restore(version):
+	"""Restore a version"""
+	version = frappe.get_doc("Version", version)
+	frappe.get_doc(version.get_data()).save()
```

Comment:
So @safwansamsudeen 's PR has better restore feature. It tries to load the document in form view again so user can review and modify things before restoring :grimacing: 

This naive version also won't work always. If any validation failure occurs, end user won't have any way to bypass it. I kinda like @safwansamsudeen's version better... at least in terms of design.